node/sc/multi_bridge_test.go

```diff
@@ -354,3 +359,651 @@ func TestMultiBridgeKLAYTransfer3(t *testing.T) {
+	info.sim.Commit()
+	assert.Nil(t, bind.CheckWaitMined(info.sim, tx))
+
+	nonceOffset := uint64(17)
+	sentNonce := nonceOffset
+	transferAmount := uint64(100)
+	sentBlockNumber := uint64(100000)
+
+	tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+	info.sim.Commit()
+	assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+	sentNonce = sentNonce + 100
+	tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+	info.sim.Commit()
+	assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+	acc = info.accounts[1]
+	tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+	info.sim.Commit()
+	assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+	for {
+		select {
+		case ev := <-info.handleCh:
+			assert.Equal(t, sentNonce, ev.HandleNonce)
+			return
+		case err := <-info.handleSub.Err():
+			t.Fatal("Contract Event Loop Running Stop by sub.Err()", "err", err)
+		case <-time.After(timeOut):
+			t.Fatal("Contract Event Loop Running Stop by timeout")
+		}
+	}
+}
+
+// TestMultiBridgeKLAYTransferParallel checks the following:
+// - set threshold to 2.
+// - two different value transfers succeed to handle the value transfer.
+func TestMultiBridgeKLAYTransferParallel(t *testing.T) {
+	info := prepareMultiBridgeEventTest(t)
+	to := common.Address{100}
+	acc := info.accounts[0]
+
+	opts := &bind.TransactOpts{From: acc.From, Signer: acc.Signer, GasLimit: gasLimit}
+	tx, err := info.b.SetOperatorThreshold(opts, voteTypeValueTransfer, 2)
+	assert.NoError(t, err)
+	info.sim.Commit()
+	assert.Nil(t, bind.CheckWaitMined(info.sim, tx))
+
+	nonceOffset := uint64(17)
+	sentNonce := nonceOffset
+	transferAmount := uint64(100)
+	sentBlockNumber := uint64(100000)
+
+	go func() {
+		acc := info.accounts[0]
+		tx := SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+		info.sim.Commit()
+		assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+		acc = info.accounts[1]
+		tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+		info.sim.Commit()
+		assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+	}()
+
+	go func() {
+		acc := info.accounts[2]
+		sentNonce := sentNonce + 20
+		tx := SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+		info.sim.Commit()
+		assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+		acc = info.accounts[3]
+		tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+		info.sim.Commit()
+		assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+	}()
+
+	handleCounter := 0
+	for {
+		select {
+		case _ = <-info.handleCh:
+			handleCounter++
+			if handleCounter == 2 {
+				return
+			}
+		case err := <-info.handleSub.Err():
+			t.Fatal("Contract Event Loop Running Stop by sub.Err()", "err", err)
+		case <-time.After(timeOut):
+			// expected timeout
+			return
```

Comment:
Why it doesn't raise an error?

Reply:
Because we are testing a failure case with timeout.